### PR TITLE
Update Changelog and README with Rust changes

### LINKED ISSUE #397: Update changelog to include Rust and change readme that Rust is somewhat feature complete

### PR DESCRIPTION
**Pull request by axic**
Fixes #397.

### CODE REVIEW

**Comment by jakelang** [comment]
Needs rebase

**Comment by axic** [comment]
@chfast please review and merge. This is my last PR for 6.3.

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +48,22 @@ and this project adheres to [Semantic Versioning].
...
+- `ExecutionResult` now has `success`, `failure` and `revert` helpers.
+  [[#297](https://github.com/ethereum/evmc/pull/297),
+   [#368](https://github.com/ethereum/evmc/pull/368)]
+- Introduction of `evmc-declare` crate with a procedural macro for easy VM declaration.
+  [[#262](https://github.com/ethereum/evmc/pull/262)]
+- Introduction of `ExecutionMessage` wrapper in Rust.
+  [[#324](https://github.com/ethereum/evmc/pull/324)]
+- Panics are now unwound safely in Rust.
+  [[#316](https://github.com/ethereum/evmc/pull/316)]
+- Added memory (leak) sanitizers to Rust.
```
Comment (chfast) [inline CHANGELOG.md]:
```suggestion
- Rust builds are now checked by sanitizers (address, leak, memory).
```

Personally, I do not report CI changes in changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +48,22 @@ and this project adheres to [Semantic Versioning].
...
   E.g. `evmc-vmtester ./my_vm,mode=interpreter`.
   [[#366](https://github.com/ethereum/evmc/pull/366)]
+- `ExecutionResult` now has `success`, `failure` and `revert` helpers.
+  [[#297](https://github.com/ethereum/evmc/pull/297),
+   [#368](https://github.com/ethereum/evmc/pull/368)]
+- Introduction of `evmc-declare` crate with a procedural macro for easy VM declaration.
+  [[#262](https://github.com/ethereum/evmc/pull/262)]
+- Introduction of `ExecutionMessage` wrapper in Rust.
+  [[#324](https://github.com/ethereum/evmc/pull/324)]
+- Panics are now unwound safely in Rust.
```
Comment (chfast) [inline CHANGELOG.md]:
```suggestion
- Panics are now handled safely in Rust API
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,22 @@ and this project adheres to [Semantic Versioning].
 - The `evmc-vmtester` tool received support for EVMC module configuration.
   E.g. `evmc-vmtester ./my_vm,mode=interpreter`.
   [[#366](https://github.com/ethereum/evmc/pull/366)]
+- `ExecutionResult` now has `success`, `failure` and `revert` helpers.
+  [[#297](https://github.com/ethereum/evmc/pull/297),
+   [#368](https://github.com/ethereum/evmc/pull/368)]
+- Introduction of `evmc-declare` crate with a procedural macro for easy VM declaration.
```
Comment (chfast) [inline CHANGELOG.md]:
```suggestion
- Introduction of `evmc-declare` Rust crate with a procedural macro for easy VM declaration.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -48,6 +48,22 @@ and this project adheres to [Semantic Versioning].
 - The `evmc-vmtester` tool received support for EVMC module configuration.
   E.g. `evmc-vmtester ./my_vm,mode=interpreter`.
   [[#366](https://github.com/ethereum/evmc/pull/366)]
+- `ExecutionResult` now has `success`, `failure` and `revert` helpers.
```
Comment (chfast) [inline CHANGELOG.md]:
```suggestion
- In Rust API, `ExecutionResult` now has `success`, `failure` and `revert` helpers.
```

**Comment by axic** [comment]
> I would rather created a big bold entry with information than Rust API received a lot of changes and transitioned to state where it is usable. Then all smaller item might be listed as sublist.

These are only the bigger changes, but the Rust API was usable even before, via the `evmc-sys` crate, which just exposes the raw C API. The `evmc-vm` crate which has some nicer wrappers, was usable before, but is more usable now. The `evmc-declare` create which automatically generates a complete VM wrapper was added, it simplifies coding a lot for users – with it, about 4 lines of code are needed to scaffold a VM, without it however it is about 100 lines.

**Comment by axic** [comment]
@chfast how about this version?

**Comment by chfast** [comment]
Looks good.

Maybe next time we try some tagging like:
- _C++_: Added new helper.
- _Go_: Fixed `evmc.Host`.
- _Rust_: The name of X has been changed to Y.

**Comment by axic** [comment]
Sounds good!

Squashed the commits.

**Pull request merged**